### PR TITLE
wleave: init at 0.3.0

### LINKED ISSUE #251058: Package request: wleave
**Project description**

A Wayland-native logout script written in Gtk3

**Metadata**

* homepage URL: https://github.com/AMNatty/wleave
* source URL: https://github.com/AMNatty/wleave
* license: mit
* platforms: unix, linux

### PR DESCRIPTION
**Pull request by LudovicoPiero**
https://github.com/AMNatty/wleave

Tested build and basic functionality
Closes #251058 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mfrw** [comment]
Result of `nixpkgs-review pr 251072` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>wleave</li>
  </ul>
</details>

File: pkgs/tools/wayland/wleave/default.nix [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      --bash <(cat completions/wleave.bash) \
+      --fish <(cat completions/wleave.fish) \
+      --zsh <(cat completions/_wleave)
+  '';
+
+  meta = with lib; {
+    description = "A Wayland-native logout script written in GTK3";
+    homepage = "https://github.com/AMNatty/wleave";
+    license = licenses.mit;
+    mainProgram = pname;
```
Comment (mfrw) [inline pkgs/tools/wayland/wleave/default.nix:56]: IMO this makes `grep`ing easy:

```suggestion
    mainProgram = "wleave";
```

**Comment by mfrw** [comment]
Result of `nixpkgs-review pr 251072` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>wleave</li>
  </ul>
</details>

**Review by mfrw — APPROVED** [review]

File: pkgs/tools/wayland/wleave/default.nix [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+, scdoc
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "wleave";
+  version = "0.3.0";
+
+  src = fetchFromGitHub {
+    owner = "AMNatty";
+    repo = pname;
```
Comment (natsukium) [inline pkgs/tools/wayland/wleave/default.nix:19]:
```suggestion
    repo = "wleave";
```

File: pkgs/tools/wayland/wleave/default.nix [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    rev = "${version}";
```
Comment (natsukium) [inline pkgs/tools/wayland/wleave/default.nix:20]:
```suggestion
    rev = version;
```
> Unnecessary string conversions should be avoided.
https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#syntax

**Review by natsukium — APPROVED** [review]
LGTM

**Pull request merged**